src/layer/marker/Icon.Default.js

```diff
@@ -25,26 +25,26 @@ L.Icon.Default = L.Icon.extend({
 	},
 
 	_getIconUrl: function (name) {
-		if (!L.Icon.Default.imagePath) {	// Deprecated, backwards-compatibility only
-			L.Icon.Default.imagePath = this._detectIconPath();
-		}
+		// if (!L.Icon.Default.imagePath) {	// Deprecated, backwards-compatibility only
```

Comment:
It looks like this would introduce a breaking change (setting L.Icon.Default.imagePath manually no longer supported), which I believe is unnecessary. Also, it will cause the code to detect the icon path every time an icon is changed (i.e. when a marker is created), which potentially is bad for performance.

I guess you need this change since `_getIconUrl` will be called twice and for your use case you need to return different paths for icon and shadow. However, I think you can achieve the same effect and still support manually setting `L.Icon.Default.imagePath` _and_ not do this for every call to `_getIconUrl`, but it would require some restructuring.

Also, if it's actually required to remove this support, the lines should just be removed, not commented out. But as mentioned above, I do not think removing them as is is a good idea.